src/cuga/backend/cuga_graph/policy/tool_guard/tool_guard_runtime.py

```
@@ -0,0 +1,607 @@
+            f"# Policy: {policy.name}\n"
+            f"{tool_guard.policy_code}\n"
+            f"# Assign the specific guard function for this policy\n"
+            f"{validate_alias} = {guard_func_name}\n"
+        )
+
+        guard_calls.extend([
+            "    try:",
+            f"        await {validate_alias}(api=api, args=args)",
+            "    except PolicyViolationException as e:",
+            "        error_msg = str(e)",
+            f"        # Check if error already contains policy name to avoid duplication",
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)